railties/test/commands/server_test.rb

```diff
@@ -22,6 +22,22 @@ def test_environment_without_server_option
     assert_nil options[:server]
   end
 
+  def test_handler_explicit_option
+    args = ["-x", "thin"]
+    options = parse_arguments(args)
+    assert_equal "thin", options[:server]
+  end
+
+  def test_handler_mistype
+    run_command(["-x", "tin"])
+    assert_match(/Could not find handler 'tin'. Maybe you meant 'thin' or 'cgi'/, output)
```

Comment:
How deterministic is it that `cgi` will be the other option here? I don't think we'd want intermittent test failures here.

Reply:
The handlers list is hard-coded, as we don't have a clean way to query the handlers from [rack/lib/rack/handler.rb](https://github.com/rack/rack/blob/master/lib/rack/handler.rb). If the order of the `HANDLERS` doesn't change often, the test should be stable.